### PR TITLE
Make it easier to use the new federated events API with TypeScript

### PR DESCRIPTION
**Pull request by caasi**
##### Description of change

I am following [Shukant Pal's guide](https://www.shukantpal.com/blog/pixijs/federated-events-api/) to use the new federated events API in my project.

When I add these lines:

```typescript
if (!('events' in app.renderer)) {
  app.renderer.addSystem(EventSystem, 'events')
}
```

tsc warns me that `addSystem()` is missing in the `AbstractRenderer` class. And the `Renderer` interface in `@pixi/events` is not compatible with `AbstractRenderer`.

So I make some changes to resolve them.

##### Pre-Merge Checklist

- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

**Review by tears330 — APPROVED** [review]

File: packages/core/src/AbstractRenderer.ts [hunk 1]
```diff
@@ -386,6 +387,15 @@ export abstract class AbstractRenderer extends EventEmitter
         return renderTexture;
     }
 
+    /**
+     * Adds a new system to the renderer.
+     *
+     * @param ClassRef - Class reference
+     * @param [name] - Property name for system
```
Comment (bigtimebuddy) [inline packages/core/src/AbstractRenderer.ts:394]: We don't need [name] here you can just use name. Our docs infer optional.

File: packages/core/src/Renderer.ts [hunk 2]
```diff
@@ -370,6 +370,9 @@ export class Renderer extends AbstractRenderer
 
     /**
      * Add a new system to the renderer.
+     *
+     * @extends PIXI.AbstractRenderer#addSystem
```
Comment (bigtimebuddy) [inline packages/core/src/Renderer.ts:374]: You can remove @extends flag, it's not functional here.

File: packages/canvas/canvas-renderer/src/CanvasRenderer.ts [hunk 3]
```diff
@@ -152,6 +152,16 @@ export class CanvasRenderer extends AbstractRenderer
         this.resize(this.options.width, this.options.height);
     }
 
+    /**
+     * Adds a new system to the renderer. It does nothing in the CanvasRenderer.
+     *
+     * @extends PIXI.AbstractRenderer#addSystem
```
Comment (bigtimebuddy) [inline packages/canvas/canvas-renderer/src/CanvasRenderer.ts:158]: Also, @extends isn't useful here either. Can be removed.

**Review by bigtimebuddy — APPROVED** [review]
Thanks for this, super clean, just a few minor docs notes.

File: packages/core/src/AbstractRenderer.ts (hunk 1 above)
Comment (caasi) [reply packages/core/src/AbstractRenderer.ts]: Fixed. I misunderstand what a `[]` is in JSDoc.

File: packages/core/src/Renderer.ts (hunk 2 above)
Comment (caasi) [reply packages/core/src/Renderer.ts]: Removed.

File: packages/canvas/canvas-renderer/src/CanvasRenderer.ts (hunk 3 above)
Comment (caasi) [reply packages/canvas/canvas-renderer/src/CanvasRenderer.ts]: Removed.

**Comment by bigtimebuddy** [comment]
Thank you @caasi

**Pull request merged**